tools/android/packaging/xbmc/src/org/xbmc/kodi/Main.java.in

```diff
@@ -182,6 +183,47 @@ public class Main extends NativeActivity implements Choreographer.FrameCallback
               | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
               | View.SYSTEM_UI_FLAG_FULLSCREEN
               | API_SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+
+
+      if (getPackageManager().hasSystemFeature("android.software.leanback"))
+      {
+        // Leanback
+
+        new Thread()
+        {
+          public void run()
+          {
+            try
+            {
+              sleep(10000);
+              mJsonRPC.updateLeanback(Main.this);
+            } catch (Exception e)
+            {
+              e.printStackTrace();
+            }
+          }
+        }.start();
```

Comment:
> On each activity resume you start a new thread that update leanback every 10 seconds

Nah. Those are one-shot threads (no loop) and mandatory because Android refuses to do httprequests on the main thread (even runnables don't work, as they run with original thread affinity). The 10 sec delay is for leaving Kodi time to settle on first run.